lib/util.js

```diff
@@ -6,9 +6,17 @@ const Debug = require('vm').runInDebugContext('Debug');
 const formatRegExp = /%[sdj%]/g;
 exports.format = function(f) {
   if (typeof f !== 'string') {
+    var i = 0;
+    var len = arguments.length;
     var objects = [];
-    for (var i = 0; i < arguments.length; i++) {
-      objects.push(inspect(arguments[i]));
+    for (var x = arguments[i]; i < len; x = arguments[++i]) {
```

Comment:
This would be a lot more immediately readable if you left the existing loop in tact and just added `var x = arguments[i];` in the loop body.